Scripts/script-MatchRegex.yml

```diff
@@ -6,37 +6,51 @@ script: |-
   var data = (typeof args.data) === 'string' ? args.data : JSON.stringify(args.data);
   var flags = args.flags ? args.flags : 'im';
   var r = new RegExp(args.regex, flags);
-  var m = r.exec(data);
-  if (m && m.length > 0) {
-      var val = m[0];
-      var group = parseInt(args.group);
-      if (args.group && m.length > group) {
-          val = m[group];
+  var m;
+  var vals = [];
+  while ((m = r.exec(data)) !== null) {
+      if (m && m.length > 0) {
+          var val = m[0];
+          var group = parseInt(args.group);
+          if (args.group && m.length > group) {
+              val = m[group];
+          }
+          var ec = {};
+          if (args.contextKey) {
+              ec[args.contextKey] = val;
+          }
       }
-      var ec = {};
-      if (args.contextKey) {
-          ec[args.contextKey] = val;
-      }
-      return {Type: entryTypes.note, Contents: val, ContentsFormat: formats.text, EntryContext: ec};
   }
+  if (vals.length) {
+      setContext('MatchRegex.results', vals)
+      return {Type: entryTypes.note, Contents: vals, ContentsFormat: formats.text, EntryContext: ec};
+  }
+  setContext('MatchRegex.results', vals)
   return {Type: entryTypes.note, Contents: 'Regex does not match', ContentsFormat: formats.text};
 type: javascript
 tags:
 - Utility
-comment: Extract regex data from given text - supports groups as well
+comment: Extract regex data from given text - supports groups and looping as well.
+  May need to provide 'g' in flags argument.
 enabled: true
 args:
 - name: data
   required: true
-  default: true
   description: The text date we should extract regex from
 - name: regex
   required: true
   description: The regex to match and extract
 - name: group
-  description: The matching group to return (if not provided the full match will be returned), group value should start at 1
+  description: The matching group to return (if not provided the full match will be
+    returned), group value should start at 1
 - name: contextKey
   description: The context key to populate with the result
 - name: flags
-  description: The regex flags to match. Default is - i.
+  description: The regex flags to match. Default is -i.
+  defaultValue: 
+outputs:
+- contextPath: MatchRegex.results
+  description: List of Regex matches
+  type: string
 scripttarget: 0
+runonce: false
```

Comment:
At the bottom of the file, please add a release note (without preceding tabs).
You can use the example below, or write your own (just make sure it's clear to other users):
`releaseNotes: "Added the option to return all matches."`